### ISSUE TITLE
Faster rcnn model has bad predictions

### ISSUE THREAD

**Issue by matthew-jurewicz** [pytorch/vision#1480]
After training the model until the sum of the losses is less than 0.1 for both the training and validation set, I save the models state dict. After loading the same models state dict and setting the model to eval mode, predictions on the test set have scores around 0.05 and are completely incorrect. The test set images aren't significantly different from the training or validation sets, and the test set is pre-processed and normalized in the same manner. I have tried different backbones, but nothing works.

**Comment by matthew-jurewicz** [comment]
torch.save(model.state_dict(), 'faster_rcnn.pt.tar')

model.load_state_dict(torch.load('faster_rcnn.pt.tar'))
model.eval()

**Comment by matthew-jurewicz** [comment]
I found a similar issue that was never resolved https://github.com/pytorch/vision/issues/1097#issue-464835069 . Following the mobilenet example in the docs, my model converges but the predictions are bogus.

**Comment by fmassa** [comment]
Hi,

I would recommend you testing your trained model in the training set. I believe that you have either:
- not trained the model for long enough
- do not have enough training data to generalize well enough

The detection models generally take a lot of iterations to get good performances. If you are training your model from scratch, you might just need to train it for longer.
I'd recommend you try using a pre-trained model on COCO, and finetune it on your own task first to get a first working solution.

Given that this is a modeling issue and not a problem with torchvision, I'm closing this issue, but let me know if you have further questions.
